server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -1377,19 +1376,52 @@ public GatedCloseable<IndexCommit> acquireSafeIndexCommit() throws EngineExcepti
         }
     }
 
+    /**
+     * Returns the lastest segmentInfos
+     */
+    public SegmentInfos getLatestSegmentInfos() {
+        return getEngine().getSegmentInfosSnapshot().get();
+    }
+
     /**
      * Returns the lastest Replication Checkpoint that shard received
      */
     public ReplicationCheckpoint getLatestReplicationCheckpoint() {
-        return new ReplicationCheckpoint(shardId, 0, 0, 0, 0);
+        final SegmentInfos latestSegmentInfos = getLatestSegmentInfos();
+        return new ReplicationCheckpoint(
```

Comment:
For a NRTReplicationEngine that is starting up, it will not yet have anything returned from `getLatestSegmentInfos` until the first replication event completes.  I think we need to null check here and return an "empty" ReplicationCheckpoint in that case, where gen and version could be `SequenceNumbers.NO_OPS_PERFORMED`.